Geometry/HGCalCommonData/src/HGCalGeomRotation.cc

```diff
@@ -0,0 +1,96 @@
+#include "Geometry/HGCalCommonData/interface/HGCalGeomRotation.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+void HGCalGeomRotation::uvMappingFromSector0(WaferCentring waferCentring,
+                                             int& moduleU,
+                                             int& moduleV,
+                                             unsigned sector) const {
+  if (sector == 0) {
+    return;
+  }
+
+  int offset;
+
+  if (waferCentring == WaferCentred) {
+    offset = 0;
+  } else if (waferCentring == CornerCentredY) {
+    offset = -1;
+  } else if (waferCentring == CornerCentredMercedes) {
+    offset = 1;
+  } else {
+    throw cms::Exception("RotationException")
```

Comment:
Exceptions are a very severe response to an error. They can terminate a workflow. Wouldn't `LogError` be a better response?

Reply:
These are cases that should never arise, so if they did it would be a bad usage of the function, and the results might be meaningless.

Reply:
OK, since the `enum class` will limit the values, this case can never occur, unless more value are added to the `enum`.